### PR TITLE
Open and bid

### PR DESCRIPTION
**Pull request by alexvandesande**
Now you can open and bid in one single auction. Also, all new bids also open 3 random auctions, this increases privacy when bidding on names that are already open

### CODE REVIEW

File: app/imports/templates/components/newBid.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<template name="components_newBid">
+  <form class="new-bid">
+      <br><h3>How much you value this name?!</h3>
```
Comment (Arachnid) [inline app/imports/templates/components/newBid.html]: I think just `?` is probably enough?!

File: app/imports/templates/components/newBid.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
+<template name="components_newBid">
+  <form class="new-bid">
+      <br><h3>How much you value this name?!</h3>          
+      <p>
+        Send the maximum amount of value you would be willing to lock for this name:<br>
+        <input type="text" min="0.01" step="any" name="bidAmount" placeholder="0.01" class="bid-amount dapp-large" pattern="[0-9\.,]*">
+      </p>
+      <br>
+      <h3>Protect Bid anonimity</h3>
```
Comment (Arachnid) [inline app/imports/templates/components/newBid.html]: s/anonimity/anonymity/

File: app/imports/templates/components/newBid.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+import { updatePendingBids } from '/imports/lib/bids';
+var template;
+
+Template['components_newBid'].onCreated(function() {
+  template = this;
+  TemplateVar.set(template, 'anonymizer', 0.5);
+  let launchRatio = (Date.now()/1000 - registrar.registryStarted.toFixed())/(8*7*24*60*60);
+  let name = Session.get('searched');
+
+  // console.log('registryStarted', launchRatio, registrar.registryStarted.toFixed());
```
Comment (Arachnid) [inline app/imports/templates/components/newBid.js]: Stray commented lines here.

File: app/imports/templates/status/open.html [hunk 4]
```diff
@@ -10,18 +10,20 @@ <h1 class="domain-name"><strong>{{name}}</strong> is available</h1>
...
+          <p>
+          Or..
+          </p>
+
+          <button type="submit" class="open-auction" disabled={{opening}}> 
             {{#if opening}}
               Opening...
             {{else}}
-              Open auction
+              Just open, don't bid now
```
Comment (Arachnid) [inline app/imports/templates/status/open.html]: ???

File: app/imports/templates/status/open.js [hunk 5]
```diff
@@ -1,41 +1,40 @@
...
+  template = this;
+  var launchRatio = (Date.now()/1000 - registrar.registryStarted.toFixed())/(8*7*24*60*60);
 
-    // console.log('registryStarted', launchRatio, registrar.registryStarted.toFixed());
-    // The goal here is to obscure the names we actually want
-    function randomName() {
-      // gets a random name from our preimage hash
-      return '0x' + web3.sha3(knownNames[Math.floor(Math.random()*knownNames.length*launchRatio)]).replace('0x','');
-    }
+  // console.log('registryStarted', launchRatio, registrar.registryStarted.toFixed());
```
Comment (Arachnid) [inline app/imports/templates/status/open.js]: More stray comments.

File: app/imports/templates/status/open.js [hunk 6]
```diff
@@ -1,41 +1,40 @@
...
+    return '0x' + web3.sha3(knownNames[Math.floor(Math.random()*knownNames.length*launchRatio)]).replace('0x','');
+  }
 
-    function randomHash() {
-      // gets a random hash
-      var randomHex = new BigNumber('0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff')
-            .times(launchRatio.toString())
-            .times(Math.random().toString())
-            .floor().toString(16);
+  function randomHash() {
```
Comment (Arachnid) [inline app/imports/templates/status/open.js]: This is duplicated in newbid; could we deduplicate it?

File: app/imports/templates/status/open.js [hunk 7]
```diff
@@ -1,41 +1,40 @@
...
-      if (availableNow.length > 2) {
-        return '0x' + web3.sha3(availableNow[Math.floor(Math.random()*availableNow.length)]).replace('0x','')
-      } else {
-        return Math.random() > 0.5 ? randomHash() : randomName();
-      }
+  function randomMix() {
+    // gets a random name we know about
+    var availableNow = _.pluck(Names.find({mode:'open', name:{$not: name}, availableDate: {$lt: Date.now()/1000}}).fetch(), 'name');
+    if (availableNow.length > 10) {
+      return '0x' + web3.sha3(availableNow[Math.floor(Math.random()*availableNow.length)]).replace('0x','')
```
Comment (Arachnid) [inline app/imports/templates/status/open.js]: Doesn't `web3.sha3` always prefix with '0x'?

File: app/imports/templates/components/newBid.html (hunk 1 above)
Comment (alexvandesande) [reply app/imports/templates/components/newBid.html]: Ops you're right! I added that as a test.

File: app/imports/templates/status/open.html (hunk 4 above)
Comment (alexvandesande) [reply app/imports/templates/status/open.html]: This is just the old button. Why not leave the option available?

File: app/imports/templates/status/open.js (hunk 7 above)
Comment (alexvandesande) [reply app/imports/templates/status/open.js]: Not in my experience. Not sure why

File: app/imports/templates/status/open.html (hunk 4 above)
Comment (Arachnid) [reply app/imports/templates/status/open.html]: Oh, now I understand the text; I thought it was informational ("It was just opened, don't bid now"). Perhaps "Open without bidding" instead?

File: app/imports/templates/components/newBid.html [hunk 8]
```diff
@@ -1,12 +1,12 @@
 <template name="components_newBid">
   <form class="new-bid">
-      <br><h3>How much you value this name?!</h3>          
+      <br><h3>How much you value this name?</h3>
```
Comment (Arachnid) [inline app/imports/templates/components/newBid.html]: s/you/do you/

File: app/imports/templates/components/newBid.html [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
   <form class="new-bid">
-      <br><h3>How much you value this name?!</h3>          
+      <br><h3>How much you value this name?</h3>          
       <p>
         Send the maximum amount of value you would be willing to lock for this name:<br>
         <input type="text" min="0.01" step="any" name="bidAmount" placeholder="0.01" class="bid-amount dapp-large" pattern="[0-9\.,]*">
       </p>
       <br>
-      <h3>Protect Bid anonimity</h3>
+      <h3>Protect Bid anonymity</h3>
```
Comment (Arachnid) [inline app/imports/templates/components/newBid.html]: Either s/a/A/ or s/B/b/ :)

**Comment by alexvandesande** [comment]
updated with comments

**Pull request merged**